### PR TITLE
feat(composer): Use centralized docker execution

### PR DESCRIPTION
**Pull request by zharinov**
ref #2863

### CODE REVIEW

File: lib/manager/composer/artifacts.ts [hunk 1]
```diff
@@ -97,28 +97,28 @@ export async function updateArtifacts(
...
-      const envVars = ['COMPOSER_CACHE_DIR'];
-      cmd += envVars.map(e => `-e ${e} `);
-      cmd += `-w "${cwd}" `;
-      cmd += `renovate/composer composer`;
+      execOptions.docker = {
+        image: 'renovate/composer',
+        tag: null,
+        dockerUser: config.dockerUser,
+        volumes: [config.localDir, process.env.COMPOSER_CACHE_DIR],
+        envVars: ['COMPOSER_CACHE_DIR'],
```
Comment (rarkins) [inline lib/manager/composer/artifacts.ts]: Are we at risk of mismatches between `execOptions.env` and `execOptions.docker.envVars`? I guess there are times we might want to pass an env variable to child_process but not to `docker run`, but I can't think of any. Can you check if we have any use cases yet? If not then we could combine the two.

Comment (zharinov) [reply lib/manager/composer/artifacts.ts]: I'll write more mocks handling env-vars machinery and the combine these parameters

**Comment by ViceIce** [comment]
needs deconflicting

File: lib/manager/composer/artifacts.ts [hunk 2]
```diff
@@ -96,28 +96,28 @@ export async function updateArtifacts(
       await fs.outputFile(localAuthFileName, JSON.stringify(authJson));
     }
     const startTime = process.hrtime();
-    let cmd: string;
+    const cmd = 'composer';
+    const execOptions: ExecOptions = {
+      cwd,
+      env,
+      inheritEnvVars: ['COMPOSER_CACHE_DIR'],
```
Comment (zharinov) [inline lib/manager/composer/artifacts.ts]: @rarkins I think I would need to make `inheritEnv` instead, which can be Object like `env`, but filled with fallback values for missing `process.env` keys.

Comment (zharinov) [reply lib/manager/composer/artifacts.ts]: In this case we'll use it instead of "raw" `env` which still can be used to "force" some variables

Comment (zharinov) [reply lib/manager/composer/artifacts.ts]: Or maybe it's okay to have `process.env` vars priority over every `env` value

Comment (rarkins) [reply lib/manager/composer/artifacts.ts]: Does it make sense if we call it something like `extraEnvVars` ?

**Comment by zharinov** [comment]
Ready for review

**Review by viceice — DISMISSED** [review]

**Comment by rarkins** [comment]
I think the cache master volume should already be mapped by the util function so not necessary here

**Pull request merged**